GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1551 @@
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError Deleting User {0}", Program.ExtractErrorMessage(e));
+                }
+            }
+
+            #endregion
+        }
+
+        private static async Task DeleteGroup(Group newGroup)
+        {
+            #region Delete Group
+
+            //*********************************************************************************************
+            // Delete the Group that we just created
+            //*********************************************************************************************
+            if (newGroup.ObjectId != null)
+            {
+                try
+                {
+                    await newGroup.DeleteAsync();
+                    Console.WriteLine("\nGroup {0} was deleted", newGroup.DisplayName);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError Deleting Group {0}", Program.ExtractErrorMessage(e));
+                }
+            }
+
+            #endregion
+        }
+
+        private static async Task DeleteApplication(Application newApp, ExtensionProperty newExtension = null)
+        {
+            #region Delete Application
+
+            //*********************************************************************************************
+            // Delete Application Objects
+            //*********************************************************************************************
+            if (newApp.ObjectId != null)
+            {
+                try
+                {
+                    if (newExtension != null)
+                    {
+                        try
+                        {
+                            newApp.ExtensionProperties.Remove(newExtension);
+                            Console.WriteLine("\nDeleted extension property: " + newExtension.Name);
+                        }
+                        catch (Exception e)
+                        {
+                            Program.WriteError("\nError deleting extension property: {0}",
+                                Program.ExtractErrorMessage(e));
+                        }
+                    }
+                    await newApp.DeleteAsync();
+                    Console.WriteLine("\nDeleted Application object: " + newApp.ObjectId);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError deleting Application: {0}", Program.ExtractErrorMessage(e));
+                }
+            }
+
+            #endregion
+        }
+
+        private static async Task DeleteServicePrincipalAndPermission(
+            ServicePrincipal servicePrincipal, 
+            OAuth2PermissionGrant permissionGrant = null)
+        {
+            try
+            {
+                await servicePrincipal.DeleteAsync();
+                Console.WriteLine("Delet
```

Comment:
ok

---
In reply to: [81250650](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r81250650) [](ancestors = 81250650)